### PR TITLE
pythonPackages.docplex: init at 2.12.182

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change

Add dependency for upcoming package qiskit-aqua #78772 @jonringer 
Depends on #78874 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python3Packages.docplex

**Comment by drewrisinger** [comment]
build fails b/c missing docloud. PR #78874 must be approved first

**Comment by drewrisinger** [comment]
Should work now that #78874 has been approved. Builds locally.

File: pkgs/development/python-modules/docplex/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    inherit pname version;
+    sha256 = "aaf150b06d44f07639aca48be1fca69c7732d57507e6adc4e8451c7a93489116";
+  };
+
+  propagatedBuildInputs = [
+    docloud
+    requests
+  ]
+    ++ lib.optional isPy27 futures
+  ;
```
Comment (jonringer) [inline pkgs/development/python-modules/docplex/default.nix]: feels odd to have these on 3 extra lines
```
    ++ lib.optional isPy27 futures;
```
or
```
  ]  ++ lib.optional isPy27 futures;
```

**Comment by jonringer** [comment]
otherwise LGTM

```
[5 built, 3 copied (0.4 MiB), 0.1 MiB DL]
https://github.com/NixOS/nixpkgs/pull/78875
3 package built:
python27Packages.docplex python37Packages.docplex python38Packages.docplex
```

**Comment by drewrisinger** [comment]
@GrahamcOfBorg build python37Packages.docplex python38Packages.docplex

**Pull request merged**